components/search/src/types.rs

```diff
@@ -0,0 +1,197 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/// The list of possible application names that are currently supported.
+#[derive(Debug, uniffi::Enum)]
+pub enum SearchApplicationName {
+    Firefox = 1,
+    FirefoxAndroid = 2,
+    FocusAndroid = 3,
+    FirefoxIOS = 4,
+    FocusIOS = 5,
+}
```

Comment:
Question 1: 
Are these constants in Rust? I wonder if you're writing constants variables here, and if they should be all caps? Or maybe that's just a javascript thing?

Question 2: 
I'm not sure if ordering matters? The list is in a different order than the [applications section for search config v2 schema ](https://searchfox.org/mozilla-central/rev/7e0ae4372c52b8183d1178132dd6493edb576738/toolkit/components/search/schema/search-config-v2-schema.json#110-116)

Reply:
> Are these constants in Rust? 

They are! Specifically, they’re enum variants with explicit discriminants, so they’ll have stable raw values when exposed to foreign languages. This makes it easier to convert them [to and from raw values in Swift and Kotlin](https://mozilla.github.io/uniffi-rs/0.27/proc_macro/index.html#variant-discriminants)—but I don’t think the Gecko JS bindings don’t do anything special with them just yet.

> I wonder if you're writing constants variables here, and if they should be all caps? Or maybe that's just a javascript thing?

Yep, it’s a JavaScript thing—Rust convention is to write enum variants in `PascalCase`. UniFFI will take care of turning them into `SHOUTY_SNAKE_CASE` when it generates the JavaScript and Kotlin bindings (and `camelCase` when it generates the Swift bindings).

Reply:
> I'm not sure if ordering matters?

With explicit discriminants, order doesn’t matter.